nntrainer/layers/conv2d_layer.cpp

```diff
@@ -237,11 +247,19 @@ void Conv2DLayer::calcDerivative(sharedConstTensors derivatives) {
 
   for (unsigned int b = 0; b < in_dim.batch(); ++b) {
     Tensor inSub = derivative.getBatchSlice(b, 1);
+    std::vector<float> in_col;
+    in_col.resize(kdim.getFeatureLen() * input_dim_padded.width() *
+                  input_dim_padded.height());
 
+    Tensor in_padded = zero_pad(0, inSub, same_pad.data());
     status =
-      conv2d_gemm(imKernel_raw, kdim, inSub, input_dim_padded, stride, same_pad,
-                  ret.getAddress(b * ret.getDim().getFeatureLen()),
-                  input_dim_padded.getFeatureLen(), true);
+      im2col(in_padded, kdim, in_col.data(), input_dim_padded, stride, true);
+    if (status != ML_ERROR_NONE)
```

Comment:
I think you'd better to add `likely` macro inside a loop (in another PRs)

Reply:
I will. thanks.